woocommerce_fusion/tasks/sync_sales_orders.py

```diff
@@ -606,11 +606,11 @@ def create_missing_items(self, items_list, woocommerce_site):
 				filters={"woocommerce_id": item_woo_com_id, "woocommerce_server": woocommerce_site},
 				fields=["parent"],
 			)
-			found_item = frappe.get_doc("Item", item_codes[0].parent) if item_codes else None
+			found_item = frappe.get_doc("Item", item_woo_com_id) if item_codes else None
```

Comment:
`item_data.get("product_id")` is not the SKU.
Product id is stored in item master in the `Item WooCommerce Server` child table
![Screenshot 2024-04-05 at 06-13-24 Bag Filter - 1 micron (lock) - woocommerce - 1872](https://github.com/tundebabzy/woocommerce_fusion/assets/818803/c9a659e9-db75-4581-8128-a059994c6af8)

Is is also used as their item code.
![Screenshot 2024-04-05 at 06-17-58 Bag Filter - 1 micron (lock) - woocommerce - 1872](https://github.com/tundebabzy/woocommerce_fusion/assets/818803/cfbe903d-b689-4db4-a6e0-f1b9a3a9f897)


I checked the woocommerce order records and I can see an `sku` property
![Screenshot 2024-04-05 at 06-12-12 catalog daicowater com~9563](https://github.com/tundebabzy/woocommerce_fusion/assets/818803/08e2198c-ed70-49f3-b677-8473f4c5658a)

That said, like we discussed, make this optional to the user. This will make it easier for us to offer this as a contribution to the original app because we are sure we will not disrupt old users. Some new users may also prefer the way the app is currently do it. 

Add a section to `Woocommerce Settings` called `Item naming settings`. In the section, add a field called `Name by` and the options would be:
- Product ID (default)
- SKU

You can now tweak the code to do something like:
```
item_woo_com_id = cstr(item_data.get("product_id")) if settings.name_by.lower() == "product id" else cstr(item_data.get("sku"))
...
...
item_name = None
if item_codes:
    item_name = item_codes[0].parent if settings.name_by.lower() == "product id" else item_woo_com_id
if item_name:
    found_item = frappe.get_doc("Item", item_name)
```
Something like that